### PR TITLE
Allow measures to specify dummy data additional population constraint

### PR DESCRIPTION
**Pull request by rebkwok**

### CODE REVIEW

File: ehrql/query_language.py [hunk 1]
```diff
@@ -50,6 +50,23 @@ class DummyDataConfig:
...
+            validate_patient_series_type(
+                self.additional_population_constraint,
+                types=[bool],
+                context="additional population constraint",
+            )
+            self.additional_population_constraint = (
+                self.additional_population_constraint._qm_node
+            )
+        else:
+            self.additional_population_constraint = None
```
Comment (madwort) [inline ehrql/query_language.py:64]: do we need this branch? (the other branch tested for `is not None`)

File: ehrql/query_language.py [hunk 2]
```diff
@@ -50,6 +50,23 @@ class DummyDataConfig:
...
+    def validate_additional_population_constraint(self):
+        if self.additional_population_constraint is not None:
+            validate_patient_series_type(
+                self.additional_population_constraint,
+                types=[bool],
+                context="additional population constraint",
+            )
+            self.additional_population_constraint = (
+                self.additional_population_constraint._qm_node
+            )
```
Comment (evansd) [inline ehrql/query_language.py:62]: I don't think this is quite right because now the "validate" method is changing the value of `self.additional_population_constraint`. So, for example, it will fail if you call it twice in a row.

I think you want to make it a `set_additional_population_constraint()` method which takes the value, validates it, and then updates the attribute if it's valid.

**Review by evansd — APPROVED** [review]
Thanks for doing this!

File: ehrql/query_language.py (hunk 2 above)
Comment (rebkwok) [reply ehrql/query_language.py:62]: Huh that's how I did it first. Will change it back, hadn't considered calling it twice
